src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -201,8 +453,259 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
+    }
+
+    /**
+     * Writes the current {@link SimpleQueryStringBuilder} object to the given
+     * output stream.
+     * 
+     * @param out The stream to write the object to.
+     **/
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(queryText);
+        out.writeFloat(boost);
+        out.writeInt(fieldsAndWeights.size());
+        for (Map.Entry<String, Float> entry : fieldsAndWeights.entrySet()) {
+            out.writeString(entry.getKey());
+            out.writeFloat(entry.getValue());
+        }
+        out.writeInt(flags);
+        out.writeOptionalString(analyzer);
+        out.writeBoolean(defaultOperator != null);
+        if (defaultOperator != null) {
+            switch (defaultOperator) {
+            case OR:
+                out.writeVInt(1);
+                break;
+            case AND:
+                out.writeVInt(2);
+                break;
+            }
+        }
+        out.writeOptionalBoolean(settings.lowercaseExpandedTerms());
+        out.writeOptionalBoolean(settings.lenient());
+        out.writeOptionalBoolean(settings.analyzeWildcard());
+        out.writeBoolean(settings.locale() != null);
+        if (settings.locale() != null) {
+            out.writeString(settings.locale().toLanguageTag());
+        }
+        out.writeOptionalString(queryName);
+        out.writeOptionalString(minimumShouldMatch);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        if (settings.locale != null) {
+            return Objects.hash(getClass(), fieldsAndWeights, analyzer, defaultOperator, queryText, queryName, minimumShouldMatch,
+                    settings.lowercaseExpandedTerms, settings.lenient, settings.analyzeWildcard, settings.locale.toLanguageTag());
+        }
+        return Objects.hash(getClass(), fieldsAndWeights, analyzer, defaultOperator, queryText, queryName, minimumShouldMatch,
+                settings.lowercaseExpandedTerms, settings.lenient, settings.analyzeWildcard, settings.locale);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        SimpleQueryStringBuilder other = (SimpleQueryStringBuilder) obj;
+        return Objects.equals(fieldsAndWeights, other.fieldsAndWeights) && Objects.equals(analyzer, other.analyzer)
+                && Objects.equals(defaultOperator, other.defaultOperator) && Objects.equals(queryText, other.queryText)
+                && Objects.equals(queryName, other.queryName) && Objects.equals(minimumShouldMatch, other.minimumShouldMatch)
+                && Objects.equals(settings, other.settings);
+    }
+
+    /**
+     * Class encapsulating the settings for the SimpleQueryString query, with
+     * their default values
+     */
+    public static class Settings {
```

Comment:
was this class simply moved down or changed too? the diff is hard to read, maybe we can leave it where it was for now to simplify this change?